presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java

```diff
@@ -181,7 +182,8 @@ public void testUnmatchedDynamicFilter()
                                 ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
                                 tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey")),
                                 filter(
-                                        "LINEITEM_OK > 0",
+                                        expression("LINEITEM_OK > 0"),
+                                        TRUE_LITERAL,
```

Comment:
This is good. But it seems too easy to use a `filter` overload which does not check for dynamic filters.

ideas

- this class could define `filter` method that delegates to `PlanMatchPattern#filter` making sure the lax matching is not used in this class
- the `PlanMatchPattern#filter`  lax overload could be renamed to eg `staticFilter`, making it easier and more tempting to use the more strict overload

Reply:
> this class could define filter method that delegates to PlanMatchPattern#filter making sure the lax matching is not used in this class

that makes sense